src/lib/builtins/mod.rs

```diff
@@ -559,6 +559,12 @@ fn builtin_not(args: &[&str], shell: &mut Shell) -> i32 {
     if check_help(args, MAN_NOT) {
         return SUCCESS;
     }
+    if args.len() < 2 {
```

Comment:
this logic seems like it could be made into a method? As its more or less the same code except for the name of the arg. I could be wrong though. Thoughts?

Reply:
it does seem a bit complicated. I copied the template from `builtin_bool`. I'm not really sure what the need is for the locking. I'm now noticing most of the other functions just use `eprintln`. I'll simplify the block but for such short functions I lean aesthetically towards leaving it inline.